src/components/computed-properties.vue

```diff
@@ -195,7 +198,28 @@ export default {
     },
     'add.property': {
       handler() {
-        this.validateDuplicate();
+        if (this.$refs.propName && this.$refs.propName.validator) {
```

Comment:
Using a ref to reach into a component's internal state is a code smell.

It's hard to read: I have to now look into the `form-input` component to get a complete picture of how this code is suppose to work.

It's fragile: If the component the ref is referring to is refactored, then this code might break; it introduces a tight coupling.

There has to be a better way to handle this without introducing this kind of tech debt. What do you think?

Reply:
yes, I changed to add validation rule with pull request in  project vue-form-elements

required ProcessMaker/vue-form-elements#64